### PR TITLE
PortManager: start at port 29100

### PR DESCRIPTION
**Pull request by brong**
This is higher and less likely to clash with other things

### CODE REVIEW

**Comment by rsto** [comment]
I'm going to check this on my VM in a bit, but it will take a while.

**Comment by ksmurchison** [comment]
We should also change the comment on the default value for `base_port` in `cassandane.ini.example`

**Comment by rsto** [comment]
Tests passed. Way better than it was before.

**Review by rsto — APPROVED** [review]
LGTM

**Review by elliefm — CHANGES_REQUESTED** [review]
Tested, works fine.

Agreed with Ken's comment though: please update the other `9100`s too... `ack 9100` finds these:

```
cassandane/cassandane.ini.example
74:##base_port = 9100

cassandane/genmail3.pl
66:        port => 9100,

cassandane/imap-append.pl
59:        port => 9100,
```

The latter two look like utility scripts which are assuming the default is 9100, but that assumption will no longer hold, and they'll break if used.  I don't think we use them, but we can save our future selves a few minutes of confusion if we ever do, by keeping their assumptions updated.

`ack '\b91\d\d\b'` additionally finds these ones:

```
cassandane/Cassandane/Test/MessageStoreFactory.pm
94:    $ms = Cassandane::MessageStoreFactory->create(uri => 'imap://victoria:secret@foo.com:9143/inbox.foo');
99:    $self->assert($ms->{port} == 9143);
102:    $ms = Cassandane::MessageStoreFactory->create(uri => 'imap://victoria@foo.com:9143/inbox.foo');
107:    $self->assert($ms->{port} == 9143);
110:    $ms = Cassandane::MessageStoreFactory->create(uri => 'imap://foo.com:9143/inbox.foo');
115:    $self->assert($ms->{port} == 9143);
```

... though I don't think these ones actually start a service on that port, they're just testing that the object is created correctly, so they won't actually clash with anything nor care that the default range has changed.

File: cassandane/Cassandane/PortManager.pm [hunk 1]
```diff
@@ -58,7 +58,7 @@ sub alloc
             if (defined($workerid) && $workerid eq 'invalid');
         my $cassini = Cassandane::Cassini->instance();
         my $cassini_base_port = $cassini->val('cassandane', 'base_port') // 0;
-        $base_port = 0 + $cassini_base_port || 9100;
+        $base_port = 0 + $cassini_base_port || 29100;
```
Comment (elliefm) [inline cassandane/Cassandane/PortManager.pm:61]: Heh... `$cassini->val()` takes an optional third argument, the default value to use if no value was configured.  So these two lines could have been one all along, like:

```
$base_port = 0 + $cassini->val('cassandane', 'base_port', '[2]9100');
```

**Review by elliefm — APPROVED** [review]

**Comment by brong** [comment]
I saw we ran into a 9100 port issue again recently, so would be great to merge this!  @elliefm I did the patch for the default value thing you suggested

File: cassandane/Cassandane/PortManager.pm (hunk 1 above)
Comment (brong) [reply cassandane/Cassandane/PortManager.pm]: Yeah!  Done :)

**Review by elliefm — APPROVED** [review]
Looks good, thanks :)

**Pull request merged**